src/modules/Feature.ts

```diff
@@ -1,10 +1,18 @@
 import WarrantClient from "../WarrantClient";
 import { CreateFeatureParams, ListFeatureOptions } from "../types/Feature";
+import { ObjectType } from "../types/ObjectType";
+import { WarrantObject } from "../types/Warrant";
 
-export default class Feature {
+export default class Feature implements WarrantObject {
```

Comment:
I think the better way to do this will be to have `WarrantObject` look like:
```typescript
interface WarrantObject {
   getObjectType(): string;
   getObjectId(): string;
}
```
and have each resource implement those methods.